pkg/clustermesh/common/interceptor.go

```diff
@@ -0,0 +1,112 @@
+// SPDX-License-Identifier: Apache-2.0
```

Comment:
file-level comment:

Is this really the right place to be doing this? I know the code is more complicated, but perhaps we should be detecting this within the etcd client Watch instead? For example, the RestartableWatchStore already has the logic to re-stale the store and generate correct deltas, even when the server is replaced?

Perhaps we just need something to tweak the watch correctly if the server ID changes there. Then it could be utilized by all etcd clients.

Thoughts?

Reply:
We could put the logic within RWS, but are there cases where that is actually useful for other etcd clients outside of ClusterMesh? It didn't seem like there was much value in making it applicable to all clients. If it won't provide value, I think having it stay in ClusterMesh is clearer on intent and poses less risk than modifying every client, since it was designed for the ClusterMesh use-case specifically

Reply:
Understood. Makes sense for now, but this is definitely some awkward code.